### PR TITLE
refactor: remove additional check on approvals

### PR DESCRIPTION
**Pull request by fubuloubu**
Removes the overly onerous approval check as a mitigation for a minor vulnerability in ERC20 implementation.

### CODE REVIEW

File: tests/functional/vault/test_shares.py [hunk 1]
```diff
@@ -187,12 +187,6 @@ def test_transferFrom(accounts, token, vault, fn_isolation):
     with brownie.reverts():
         vault.transferFrom(a, b, vault.balanceOf(a) // 2, {"from": c})
 
-    # Show that approval ordering protection works
```
Comment (Macarse) [inline tests/functional/vault/test_shares.py:190]: if the change was redundant, why did you need to remove this part of the test?

Comment (fubuloubu) [reply tests/functional/vault/test_shares.py:190]: wrong choice of words, I meant overly onerous

**Review by lbertenasco — APPROVED** [review]
This is fine, the "attack" es not an issue.
It does NOT make any sense to do this kind of checks on our contracts (where users usually approve an infinite allowance).
Also where we mostly interact between "internal" contracts by only approving the exact amount that will be utilized making the allowance 0 again.
